src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -3569,6 +3569,8 @@ static QVariant fcnGetFeatureById( const QVariantList &values, const QgsExpressi
 
     QgsFeatureRequest req;
     req.setFilterFid( fid );
+    req.setConnectionTimeout( 10000 );
```

Comment:
This doesn't actually DO anything yet. No providers retrieve the timeout or do anything with it.

Reply:
Oh wait - I see you've fixed that here.

I think you'll also need to fix test_provider_postgres.py:testTimeout  -- because it's obviously not working as is ;)

Reply:
Good point, let's do that in another PR.